offchain-modules/lib/src/dapp/indexer/eth_indexer.rs

```diff
@@ -171,17 +289,25 @@ impl EthIndexer {
             )
             .await?;
             let proof_json = serde_json::to_string(&eth_proof_json)?;
+            let recipient_lockscript = hex::encode(eth_proof_json.recipient_lockscript);
+            let lock_tx: DexLockTx = DexLockTx {
+                recipient_lockscript: recipient_lockscript.clone(),
+                expect_code_hash: self.recipient_lockscript_code_hash.clone(),
+            };
+            if !lock_tx.filter() {
+                return Ok(false);
+            }
```

Comment:
I don't think this is the correct way to use `trait`. We should combine `trait` and `generic type`, pass a generic type which impl trait to `EthIndexer struct` when new a `EthIndexer`